decode.go

```diff
@@ -243,32 +247,32 @@ func (u *unmarshaler) unmarshalScalar(
 	node *yaml.Node,
 	field protoreflect.FieldDescriptor,
 	forKey bool,
-) protoreflect.Value {
+) (protoreflect.Value, bool) {
```

Comment:
The caller previously had to check `value.IsValid()` before using the returned value. None of the callers actually did, so they all had potential panic bugs (which I encountered when trying to use a group field).

To make it more explicit and more easily require that the caller check the value, it now also returns a bool that is false when the value is invalid.